common/Fabric-Install-Utilities.psm1

```diff
@@ -580,13 +580,21 @@ function Get-Certificate($certificateThumbprint)
 }
 
 function Get-DecryptedString($encryptionCertificate, $encryptedString){
-    if($encryptedString.StartsWith("!!enc!!:")){
+    if ($encryptedString.StartsWith("!!enc!!:")) {
         $cleanedEncryptedString = $encryptedString.Replace("!!enc!!:","")
-        $clearTextValue = [System.Text.Encoding]::UTF8.GetString($encryptionCertificate.PrivateKey.Decrypt([System.Convert]::FromBase64String($cleanedEncryptedString), $true))
-        return $clearTextValue
-    }else{
+    } else{
         return $encryptedString
     }
+
+    if ($null -eq $encryptionCertificate.PrivateKey) {
+        $privateKey = [System.Security.Cryptography.X509Certificates.RSACertificateExtensions]::GetRSAPrivateKey($encryptionCertificate)
```

Comment:
Add logging here so we can track whether teh cert is CNG or CSP.

Reply:
Can do